src/sql/workbench/contrib/query/common/queryResultsDisplayStatus.ts

```diff
@@ -0,0 +1,47 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Disposable } from 'vs/base/common/lifecycle';
+import { Emitter, Event } from 'vs/base/common/event';
+
+export enum QueryResultsWriterMode {
+	ToGrid,
+	ToFile,
+}
+
+export class QueryResultsWriterStatus extends Disposable {
+	public static instance: QueryResultsWriterStatus = undefined;
+
+	private writerMode: QueryResultsWriterMode;
+
+	private readonly _onStatusChanged = this._register(new Emitter<void>());
+	public readonly onStatusChanged: Event<void> = this._onStatusChanged.event;
+
+	public static getInstance(): QueryResultsWriterStatus {
```

Comment:
why does this need to be singleton?  how does this work if multiple queries are running at the same time with different output modes?  Is there are way to more directly hook this status into the existing query execution object model?

Reply:
It currently doesn't do support multiple output modes. It goes with the output mode the user selected and uses that for all queries. Where is the query execution object model found?